lib/Doctrine/DBAL/Types/SimpleArrayType.php

```diff
@@ -47,6 +47,10 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
         if (!$value) {
             return null;
         }
+        
+        if (!is_array($value)) {
```

Comment:
I personally don't think we should allow non-array values for an array type. This looks like a misusage of the type.